projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -202,16 +260,57 @@ export class CartesianLegend {
     this.activeSeriesSubject.next();
   }
 
-  private updateActiveSeries(seriesEntry: Series<{}>): void {
-    if (this.isDefault) {
-      this.activeSeries = [];
-      this.activeSeries.push(seriesEntry);
-      this.isDefault = false;
+  private getGroupedSeries(): Series<{}>[][] {
+    if (!this.isGrouped) {
+      return [];
+    }
+
+    const nonHiddenSeries = this.series.filter(series => !series.hide);
+    const groupedSeries: Series<{}>[][] = [];
```

Comment:
What is the structure of a groupedSeries? Why is it two dimensional?

Reply:
Guessing a `Map<String, Series<TData>[]>` is a better data structure here. Looks like we're trying to get a list of series split by their groups.

Reply:
done!!